Tests/ResolutionTests/resolutionTests.swift

```diff
@@ -103,4 +234,39 @@ class resolutionTests: XCTestCase {
         }
     }
     
+    func checkError(result: Result<String, ResolutionError>, expectedError: ResolutionError)  {
+        switch result {
+        case .success:
+            XCTFail("Expected \(expectedError), but got none")
+        case .failure(let error):
+            if let catched = error as? ResolutionError {
+                assert(catched == expectedError, "Expected \(expectedError), but got \(catched)");
+                return ;
+            }
+            XCTFail("Expected ResolutionError, but got different \(error)");
+        }
+    }
+    
+}
+
+extension ResolutionError: Equatable {
+    public static func == (lhs: ResolutionError, rhs: ResolutionError) -> Bool {
+        
+        switch (lhs, rhs) {
+        case ( .UnregisteredDomain, .UnregisteredDomain):
+            return true
+        case ( .UnsupportedDomain, .UnsupportedDomain):
+            return true
+        case ( .UnconfiguredDomain, .UnconfiguredDomain):
+            return true
+        case ( .RecordNotFound, .RecordNotFound):
+            return true
+        case ( .UnsupportedNetwork, .UnsupportedNetwork):
+            return true
+        case ( .UnknownError(_), .UnknownError(_)):
+            return false
+        default:
+            return false
+        }
```

Comment:
```suggestion
        case (.UnregisteredDomain, _),
            (.UnsupportedDomain, _),
            (.UnconfiguredDomain, _),
            (.RecordNotFound, _),
            (.UnsupportedNetwork, _),
            (.UnknownError(_), _ ):
            return false
        }
```

If we do it this way, whenever we add a new variant, we won't forget to update this function (afaik, Swift compiler requires exhaustive `switch` statements)